addons/web_editor/static/src/js/wysiwyg/widgets/media.js

```diff
@@ -668,7 +668,7 @@ var FileWidget = SearchableMediaWidget.extend({
         return this._rpc({
             route: '/web_editor/attachment/add_url',
             params: {
-                'url': this.$urlInput.val(),
+                'url': this.$urlInput.val().replace(/(\.\w{3,4})\?.+/, '$1'),
```

Comment:
Right now, only URLs ending in one of those extensions are considered valid for images: `IMAGE_EXTENSIONS: ['.jpg', '.jpeg', '.jpe', '.png', '.svg', '.gif']`.
That means images such as https://www.odoo.com/web/image/18159715 are not considered valid anyway, parameters or not.
The purpose of this feature is only to handle parameters for the image in the URL, not support images without the file extension in the URL.
Following that implementation, the regex is not looking for the TLD, but for the file extension (which is either 3 or 4 characters long), since the extension has to be in the URL for it to be a valid image.

That being said, using `.split('?')` then taking the last in the list (in case there are '?' elsewhere) would indeed probably work for 99% of cases, which means that using a regex for this, while more robust, would probably be overkill.

As for keeping the parameters themselves, the spec says to "ignore parameters after [the file extension]", which can be interpreted as not keeping them. @pierre-pde could you confirm whether parameters should be kept or not?

Reply:
@SwagSamaSempai are you developer to follow line write on a spec, or make somthing that make sense ? 

Ok my bad for regexp, but proof that code was probably no so 'explicit'

You are saying that because spec is 'stupid', you develop wrong code ?
Ignore the query string to catch the extension, not to use the picture... 
Spec give you the direction, not the final point...

about `.split('?')`... no... use the first (all after is arguments) so code should be 
> url = url.split('?')[0]  and no if + split + splice + join...

Or use new Url() as suggested to handle the 100th %

Reply:
Hello

quid about this ? 

> Ignore the query string to catch the extension, not to use the picture...
Spec give you the direction, not the final point...

I see that you updated your split but not this part... not agree with it ?

Reply:
It is actually done. The reason why Odoo didn't support parameters in images is because it computes the mimetype based on the end part of the URL. If there are parameters, the end part won't be recognised, and it will thus be considered as an `application/octet-stream`.
Adding the second split in `ir_attachments.py` makes it so the end part is recognised, and the mimetype is then set to a normal value such as `image/png`.
Once the correct mimetype is set, the file src itself can contain parameters without issue.

Reply:
nice, ty. I didnt see that the split was moved guess_mimetype :+1: 

thanks